Xamarin.Forms.Core/EffectiveFlowDirection.cs

```diff
@@ -0,0 +1,13 @@
+using System;
+
+namespace Xamarin.Forms
+{
+	[Flags]
+	public enum EffectiveFlowDirection
```

Comment:
most of the implementation hide `EffectiveFlowDirection` through explicit interface implementation, so I guess this isn't mean to be used by our users at all ? if so, attribute the type or move it to the Internals namespace

Reply:
It should only be set by `Core` code (users set `FlowDirection`), but it should be exposed via `IVisualElementController` so renderers can use it.

Reply:
API only used by renderers is usually attributed, or in the internals namespace, or both

Reply:
Right, but our users will want to access this for their custom renderers, so it feels like it should be exposed.